ivy/__init__.py

```diff
@@ -202,17 +202,37 @@ def __new__(cls, shape_tup):
                 current_backend(shape_tup).NativeArray,
             )
         ivy.utils.assertions.check_isinstance(shape_tup, valid_types)
-        if isinstance(shape_tup, int):
-            shape_tup = (shape_tup,)
+        if isinstance(shape_tup, ivy.NativeShape):
+            self._shape = shape_tup
```

Comment:
regarding the transpiler I'd say this is the only relevant one, so can't say much on the other cases